### PR TITLE
Add more specific dhcpv4 options

### PR DESCRIPTION
**Pull request by get9**
This PR adds some more specific options to dhcpv4. Currently, only the ones used by the main dhcpv4 package will be added.

### CODE REVIEW

File: dhcpv4/option_class_identifier.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import (
+	"fmt"
+)
+
+// This option implements the Class Identifier option
+// https://tools.ietf.org/html/rfc2132
+
+// OptClassIdentifier represents the DHCP message type option.
+type OptClassIdentifier struct {
+	Identifier string
```
Comment (insomniacslk) [inline dhcpv4/option_class_identifier.go]: should this be `string` or `[]byte` ? Looking at the RFC I'd say `[]byte`, then all the string conversions should be done elsewhere

File: dhcpv4/option_class_identifier.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+// ToBytes returns a serialized stream of bytes for this option.
+func (o *OptClassIdentifier) ToBytes() []byte {
+	return append([]byte{byte(o.Code()), byte(o.Length())}, []byte(o.Identifier)...)
+}
+
+// String returns a human-readable string for this option.
+func (o *OptClassIdentifier) String() string {
+	return fmt.Sprintf("Class Identifier -> %v", o.Identifier)
```
Comment (insomniacslk) [inline dhcpv4/option_class_identifier.go]: e.g. if the above becomes `[]byte`, this should be converted to string first

File: dhcpv4/option_maximum_dhcp_message_size.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+type OptMaximumDHCPMessageSize struct {
+	Size uint16
+}
+
+// ParseOptMaximumDHCPMessageSize constructs an OptMaximumDHCPMessageSize struct from a sequence of
+// bytes and returns it, or an error.
+func ParseOptMaximumDHCPMessageSize(data []byte) (*OptMaximumDHCPMessageSize, error) {
+	// Should at least have code, length, and message type.
+	if len(data) < 4 {
+		return nil, errors.New("too short of byte stream")
```
Comment (insomniacslk) [inline dhcpv4/option_maximum_dhcp_message_size.go]: use `ErrShortByteStream` here

File: dhcpv4/options.go [hunk 4]
```diff
@@ -27,23 +35,26 @@ func ParseOption(data []byte) (Option, error) {
...
-	switch code {
+	switch OptionCode(data[0]) {
+	case OptionDHCPMessageType:
+		opt, err = ParseOptMessageType(data)
+	case OptionParameterRequestList:
+		opt, err = ParseOptParameterRequestList(data)
+	case OptionRequestedIPAddress:
+		opt, err = ParseOptRequestedIPAddress(data)
+	case OptionServerIdentifier:
+		opt, err = ParseOptServerIdentifier(data)
```
Comment (insomniacslk) [inline dhcpv4/options.go]: here you didn't `switch` on all the options you implemented, e.g. `OptMaximumDHCPMessageSize` and others are missing

File: dhcpv4/option_class_identifier.go (hunk 1 above)
Comment (get9) [reply dhcpv4/option_class_identifier.go]: I was going to make the specific opts more API/human-friendly, then handle serialization in the ToBytes method.

**Comment by insomniacslk** [comment]
This looks good to me. I'll wait for CI to finish the tests before squashing and merging

File: dhcpv4/option_class_identifier.go (hunk 1 above)
Comment (insomniacslk) [reply dhcpv4/option_class_identifier.go]: I think that the API should be as friendly as possible, as long as it doesn't create other kinds of risk. The string could be a multi-byte string when decoded, which may cause issues with `Length`, and the decoding of all the following options as a consequence

Comment (insomniacslk) [reply dhcpv4/option_class_identifier.go]: as you showed me offline with https://play.golang.org/p/sXdfZg0987B, your code is good, and I learned something new, thanks!

**Comment by insomniacslk** [comment]
@get9 is this ok to merge or do you plan to add more commits?

**Comment by get9** [comment]
Good to merge!

**Pull request merged**